app/controllers/api/templates_controller.rb

```diff
@@ -0,0 +1,38 @@
+class Api::TemplatesController < ApplicationController
+  before_action :find_template, only: [:update]
```

Comment:
Why the naming convention which you followed is violated here? At one place, you are using `set_` and here `find_` ?